### PR TITLE
Update stage IV and add proposal

### PR DESCRIPTION
**Pull request by JacekGlen**

### CODE REVIEW

File: docs/gas-schedule-proposal.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+Let's run through the consequences of the radical gas schedule proposal. The cheapest operations are priced at 1 gas and we gradually increase the gas cost for more complex operations. As a result most arithmetic and basic opcodes will be much cheaper, i.e. valued at 1 rather than 3 or 5. Then all other operations will be adjusted accordingly, usually by lowering the gas cost. 
+
+In this scenario the storage cost remains at the same level as this refects the network cost of storing data. Thus the radical gas schedule proposal expands the gap between the cost of storage and computation. This is a good thing as it makes it more expensive to store data than to compute it. Also the memory expansion cost remains the same to increase the security of the network.
+
+Pros:
+- The gas cost reflects the true computational cost of the operations
+- The larger gap between the cost of storage and computation promotes more efficient use of the network
+- Configurable gas schedules are easier update in the future
```
Comment (lukasz-glen) [inline docs/gas-schedule-proposal.md:62]: Configurable gas schedules? What do you mean? This therm was not used yet. Some clarification would be suitable here.

File: docs/gas-schedule-proposal.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+|  | CALL | 100 | 72 |
+|  | STATICCALL | 100 | 65 |
+|  | DELEGATECALL | 100 | 60 |
+|  | RETURN | 0 | 3 |
+|  | REVERT | 0 | 7 |
+
+
+| Precompile | Name | Current Gas | Proposed Gas |
+| ------------- | ------------- | ------------- | ------------- |
+|  | ECRECOVER | 3100 | 7064 |
```
Comment (lukasz-glen) [inline docs/gas-schedule-proposal.md:211]: Why in the radical scenario there is more precompiles to be repriced than in the conservative scenario?

File: docs/gas-schedule-proposal.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+|  | MLOAD | 3 | 2 |
+| 52 | MSTORE | static_gas = 3 <br/> dynamic_gas = memory_expansion_cost | static_gas = 2 <br/> dynamic_gas = memory_expansion_cost |
+| 53 | MSTORE8 | static_gas = 3 <br/> dynamic_gas = memory_expansion_cost | static_gas = 2 <br/> dynamic_gas = memory_expansion_cost |
+|  | JUMP | 8 | 1 |
+|  | JUMPI | 10 | 2 |
+|  | PC | 2 | 1 |
+|  | MSIZE | 2 | 1 |
+|  | GAS | 2 | 1 |
+|  | JUMPDEST | 1 | 1 |
+|  | PUSH1 | 3 | 1 |
```
Comment (lukasz-glen) [inline docs/gas-schedule-proposal.md:125]: maybe PUSHx would make it more readable?

File: docs/gas-schedule-proposal.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+## Radical Gas Schedule Proposal
+The idea behind the radical gas schedule proposal is a complete overhaul of the current gas schedule. Rather then just changing the most mispriced opcodes, we propose to change all opcodes to better reflect the computational cost of the operations.
+
+Let's run through the consequences of the radical gas schedule proposal. The cheapest operations are priced at 1 gas and we gradually increase the gas cost for more complex operations. As a result most arithmetic and basic opcodes will be much cheaper, i.e. valued at 1 rather than 3 or 5. Then all other operations will be adjusted accordingly, usually by lowering the gas cost. 
+
+In this scenario the storage cost remains at the same level as this refects the network cost of storing data. Thus the radical gas schedule proposal expands the gap between the cost of storage and computation. This is a good thing as it makes it more expensive to store data than to compute it. Also the memory expansion cost remains the same to increase the security of the network.
+
+Pros:
+- The gas cost reflects the true computational cost of the operations
```
Comment (lukasz-glen) [inline docs/gas-schedule-proposal.md:60]: "true" is too strong statement

File: docs/gas-schedule-proposal.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+
+| Precompile | Name | Current Gas | Proposed Gas |
+| ------------- | ------------- | ------------- | ------------- |
+| 01 | ECRECOVER | 3000 | 17000 |
+| 06 | ECADD | 150 | 3500 |
+| 07 | ECMUL | 6000 | 10000 |
+| 0A | POINTEVAL | 50000 | 400000 |
+
+
+## Radical Gas Schedule Proposal
```
Comment (lukasz-glen) [inline docs/gas-schedule-proposal.md:52]: There were some discussion in the community on a similar approach. Maybe we could give a reference?

File: docs/gas-schedule-proposal.md (hunk 2 above)
Comment (JacekGlen) [reply docs/gas-schedule-proposal.md:153]: The scale is different, so some of the precompiles that were previously within range now need lower cost.

**Pull request merged**